src/renderer/cluster-frame.tsx

```diff
@@ -211,6 +211,12 @@ export class ClusterFrame extends React.Component {
               {this.renderExtensionTabLayoutRoutes()}
               {this.renderExtensionRoutes()}
               <Redirect exact from="/" to={this.startUrl}/>
+              {
+                ClusterPageRegistry
+                  .getInstance()
+                  .redirectEntries
+                  .map(url => <Redirect key={url} from={url} to={this.startUrl} />)
+              }
```

Comment:
Why not to make `this.startUrl` just default page for everything that not matched specific routes above?

So I mean something like this would work? (without saving `knownUrls`)

```tsx
        // ...
        <Route component={Apps}/>
        <Route component={NotFound}/>
        {this.renderExtensionTabLayoutRoutes()}
        {this.renderExtensionRoutes()}
        <Redirect from="*" to={this.startUrl}/>
```

Reply:
Yeah that would work to, my only thought was that then we wouldn't be able to have the "NotFound" route which I figured we wanted. But if we don't then I completely agree that solution is better.

Reply:
Probably we don't need `NotFound` at all as it doesn't carry useful information for user (just a text _Page not found_). 
Better to ask others @lensapp/lens-maintainers Wdyt?

Reply:
TBH I am fine with removing `NotFound`

Reply:
I guess "page not found" is useful to make the user aware that something changed. But what @ixrock suggests is nice. Instead of the NotFound page can a notification be shown that says something like "Page not found, returning to start"?

Reply:
I will investigate if that is possible